contracts/MarketMaker.sol

```diff
@@ -54,43 +59,70 @@ contract MarketMaker is Ownable, IERC1155TokenReceiver {
         collateralToken = _collateralToken;
         conditionId = _conditionId;
         fee = _fee;
+
+        require(collateralToken.transferFrom(tx.origin, this, _funding) && collateralToken.approve(pmSystem, _funding));
+        uint[] memory partition = generateBasicPartition();
+        pmSystem.splitPosition(collateralToken, bytes32(0), conditionId, partition, _funding);
+        funding = _funding;
+
         stage = Stages.MarketCreated;
+        emit AutomatedMarketMakerCreated(funding);
     }
 
     function calcNetCost(int[] outcomeTokenAmounts) public view returns (int netCost);
 
     /// @dev Allows to fund the market with collateral tokens converting them into outcome tokens
     /// @param _funding Funding amount
-    function fund(uint _funding)
+    function changeFunding(int _funding)
         public
         onlyOwner
-        atStage(Stages.MarketCreated)
+        atStage(Stages.MarketPaused)
     {
-        // Request collateral tokens and allow event contract to transfer them to buy all outcomes
-        require(   collateralToken.transferFrom(msg.sender, this, _funding)
-                && collateralToken.approve(pmSystem, _funding));
+        // Either add or subtract funding based off whether the _funding parameter is negative or positive
+        if (_funding > 0) {
+            require(collateralToken.transferFrom(msg.sender, this, uint(_funding)) && collateralToken.approve(pmSystem, uint(_funding)));
+            uint[] memory addFundingPartition = generateBasicPartition();
+            pmSystem.splitPosition(collateralToken, bytes32(0), conditionId, addFundingPartition, uint(_funding));
+            funding = funding.add(uint(_funding));
+            emit MarketFundingChanged(_funding);
+        }
+        if (_funding < 0) {
+            uint[] memory removeFundingPartition = generateBasicPartition();
+            pmSystem.mergePositions(collateralToken, bytes32(0), conditionId, removeFundingPartition, uint(_funding));
+            funding = funding.sub(uint(_funding));
+            require(collateralToken.transfer(owner(), uint(_funding)));
```

Comment:
Should be

```solidity
// notice the negation inside of the typecast
uint(-fundingChange)
```

in the three lines above.

Note for the future: should combine `splitPosition` and `mergePositions` into one function, as code duplication causes things like this to happen.

Reply:
Why is this negation required?

Reply:
![screenshot from 2018-12-03 20-07-23](https://user-images.githubusercontent.com/298447/49413976-5ff5b400-f737-11e8-96cb-ae26943d2a42.png)